src/use-trait.js

```diff
@@ -0,0 +1,72 @@
+import { decorate } from './private/utils'
+
+const isFunction = (function ({toString}) {
+  const tag = toString.call(toString)
+  return value => toString.call(value) === tag
+})(Object.prototype)
+
+const {push: appendToArray} = Array.prototype
+
+const {
+  defineProperty,
+  getOwnPropertyNames,
+  getOwnPropertySymbols,
+  getOwnPropertyDescriptor
+} = Object
+
+function safeAssign(target, source, ignoredProperties = {}) {
+  const properties = getOwnPropertyNames(source)
+  appendToArray.apply(properties, getOwnPropertySymbols(source))
+
+  for (let i = 0, n = properties.length; i < n; ++i) {
+    const prop = properties[i]
+
+    if (ignoredProperties[prop]) {
+      continue
+    }
+
+    const descriptor = getOwnPropertyDescriptor(source, prop)
+
+    // Only assign own properties.
+    if (!prop) {
+      continue
+    }
+
+    // Do not overwrite existing (own or inherited) properties.
+    if (prop in target) {
+      throw new Error('the property ' + prop + ' already exists')
```

Comment:
I'd like a way to resolve such conflicts, instead of preventing them outright. Either order based (left to right apply), some sort of explicit resolution syntax as a second argument of the decorator, or a combination of both.

I feel like any method/prop on the target class should *always* win, silently.

Something like:

```js
const A {
  kickDog() {}
}

const B = {
  kickDog() {}
};

@useMixins(A, B, { kickDog: A })
class Person {}
```

I'm open to discussions though. Existing languages are all over the place.